### PR TITLE
[TACHYON-1139] Add unit tests for BufferedBlockOutStream

### PR DESCRIPTION
**Pull request by aaudiber**
https://tachyon.atlassian.net/projects/TACHYON/issues/TACHYON-1139

### CODE REVIEW

File: clients/unshaded/src/test/java/tachyon/client/block/TestBufferedBlockOutStream.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+ * the License.
+ */
+
+package tachyon.client.block;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+
+public class TestBufferedBlockOutStream extends BufferedBlockOutStream {
```
Comment (yupeng9) [inline clients/unshaded/src/test/java/tachyon/client/block/TestBufferedBlockOutStream.java]: If this is only used for `BufferedBlockOutStreamTest`, we only inline it in the test class.

**Comment by yupeng9** [comment]
Thanks for improving the test coverage. Maybe we can consolidate them into one file?

File: clients/unshaded/src/test/java/tachyon/client/block/TestBufferedBlockOutStream.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+  // Shouldn't need more than this for unit tests
+  private static final int MAX_DATA = 1000;
+  private ByteBuffer mDataWritten;
+  private boolean mCancelled;
```
Comment (calvinjia) [inline clients/unshaded/src/test/java/tachyon/client/block/TestBufferedBlockOutStream.java]: `mCanceled`?

File: clients/unshaded/src/test/java/tachyon/client/block/TestBufferedBlockOutStream.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  }
+
+  public boolean isClosed() {
+    return mClosed;
+  }
+
+  @Override
+  public void cancel() throws IOException {
+    mCancelled = true;
+    mClosed = true;
```
Comment (calvinjia) [inline clients/unshaded/src/test/java/tachyon/client/block/TestBufferedBlockOutStream.java]: This should probably call `close()`?

File: clients/unshaded/src/test/java/tachyon/client/block/TestBufferedBlockOutStream.java (hunk 1 above)
Comment (calvinjia) [inline clients/unshaded/src/test/java/tachyon/client/block/TestBufferedBlockOutStream.java]: Although this is only for testing, could you add some javadocs to it so its clearer what others can expect from using this class?

Comment (aaudiber) [inline clients/unshaded/src/test/java/tachyon/client/block/TestBufferedBlockOutStream.java]: I have a couple of commits staged on top of this one, and they use this file.

File: clients/unshaded/src/test/java/tachyon/client/block/TestBufferedBlockOutStream.java (hunk 2 above)
Comment (aaudiber) [inline clients/unshaded/src/test/java/tachyon/client/block/TestBufferedBlockOutStream.java]: Done

File: clients/unshaded/src/test/java/tachyon/client/block/TestBufferedBlockOutStream.java (hunk 3 above)
Comment (aaudiber) [inline clients/unshaded/src/test/java/tachyon/client/block/TestBufferedBlockOutStream.java]: Done

File: clients/unshaded/src/test/java/tachyon/client/block/TestBufferedBlockOutStream.java (hunk 1 above)
Comment (aaudiber) [inline clients/unshaded/src/test/java/tachyon/client/block/TestBufferedBlockOutStream.java]: Added some javadocs

**Comment by yupeng9** [comment]
LGTM

**Comment by calvinjia** [comment]
LGTM

**Pull request merged**